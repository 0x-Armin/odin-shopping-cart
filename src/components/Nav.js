import React, { createFactory, useEffect, useState } from "react";
import { Link, useSearchParams } from "react-router-dom";

import "../style/reset.css";
import "../style/style.css";

const Nav = ({cart}) => {
  const [numCartItems, setNumCartItems] = useState(0);

  const countCartItems = (cart) => {
    let sum = 0;
    cart.forEach((v) => sum += v);

    return sum;
  }

  useEffect(() => {
    setNumCartItems(countCartItems(cart));
  }, [cart]);

  return (
    <nav className="navigation-bar">
      <h3 className="logo">SNEAKERS UNDERGROUND</h3>
      <ul className="navigation-links">
        <Link to="/">
          <li>Home</li>
        </Link>
        <Link to="/shop">
          <li>Shop</li>
        </Link>
        <Link to="/cart">
          <li>Cart ({numCartItems})</li>
        </Link>
      </ul>
    </nav>
  );
};

export default Nav;
