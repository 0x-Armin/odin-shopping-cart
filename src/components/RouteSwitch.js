import React, { useState, useEffect } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import Nav from './Nav';
import Home from './Home';
import Shop from './Shop';
import Cart from './Cart';

const RouteSwitch = () => {
  const [cart, setCart] = useState(new Map());

  return (
    <BrowserRouter>
      <Nav cart={cart} />
      <Routes>
        <Route path='/' element={<Home />} />
        <Route path='/shop' element={<Shop cart={cart} setCart={setCart} /> } />
        <Route path='/cart' element={<Cart cart={cart} setCart={setCart}/>} />
      </Routes>
    </BrowserRouter>
  )
}

export default RouteSwitch;