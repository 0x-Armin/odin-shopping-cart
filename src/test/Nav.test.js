import React from "react";
import { render } from "@testing-library/react";

import Nav from "../components/Nav";
import RouteSwitch from "../components/RouteSwitch";

describe("Nav component", () => {
  it("renders correct logo", () => {
    const { getByRole } = render(<Nav />, {wrapper: RouteSwitch});
    expect(getByRole("heading").textContent).toMatch(/SNEAKERS UNDERGROUND/i);
  });

  it("check 3 li items in nav", () => {
    const { getAllByRole } = render(<Nav />, {wrapper: RouteSwitch});
    expect(getAllByRole('listitem')).toHaveLength(3);
  })

  it("renders home page with sample sneakers", () => {
    const { container } = render(<Nav />, {wrapper:RouteSwitch});
    expect(container).toMatchSnapshot();
  })
});
