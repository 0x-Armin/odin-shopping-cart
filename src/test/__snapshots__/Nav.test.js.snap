// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders home page with sample sneakers 1`] = `
<div>
  <nav
    class="navigation-bar"
  >
    <h3
      class="logo"
    >
      <a
        href="/"
      >
        SNEAKERS UNDERGROUND
      </a>
    </h3>
    <ul
      class="navigation-links"
    >
      <a
        href="/"
      >
        <li>
          Home
        </li>
      </a>
      <a
        href="/shop"
      >
        <li>
          Shop
        </li>
      </a>
      <a
        href="/cart"
      >
        <li>
          Cart (
          0
          )
        </li>
      </a>
    </ul>
  </nav>
  <div
    class="home-page"
  >
    <div
      class="sneakers-row"
    >
      <img
        alt="New Balance BB550 Sneaker"
        class="sneaker-home"
        src="newbalance-bb550.webp"
      />
      <img
        alt="Reebok Classic Sneaker"
        class="sneaker-home"
        src="reebok-classic.webp"
      />
      <img
        alt="Mizuno Waverider 10 Sneaker"
        class="sneaker-home"
        src="mizuno-waverider10.webp"
      />
    </div>
    <div
      class="home-stmt"
    >
      Build your sneakers playground
and your reputation renowned
    </div>
    <div
      class="home-cta"
    >
      Shop 
      <u>
        <a
          href="/shop"
        >
          now
        </a>
      </u>
    </div>
  </div>
</div>
`;
